### ISSUE TITLE
Incorrect padding exception thrown when DecodeBase64String is called in base.py

### ISSUE THREAD

**Issue by adisaraf** [Azure/azure-cosmos-python#136]
While running the azure cli for cosmosdb I get an error `Incorrect Padding`. See below:

```
az cosmosdb database show --db-name <db-name> --key <secondary key you provided> --url-connection <url>

Incorrect padding

Traceback (most recent call last):

  File "/usr/lib64/az/lib/python2.7/site-packages/knack/cli.py", line 194, in invoke
```
I was able to fix this problem by tracking it down to the `base.py` file which contains a method called `DecodeBase64String`. This method fails with that error when `base64.standard_b64decode` is called. I added `==` at the end of the string to be decoded and that appears to have fixed the problem. However, I am not sure if this is the correct solution. 

```
def DecodeBase64String(string_to_decode):
    """Decodes a Base64 encoded string by replacing '-' with '/'

    :param string string_to_decode:
        String to decode.

    :return:
        Path with beginning and ending slashes trimmed.
    :rtype: str
    """
    # '-' is not supported char for decoding in Python(same as C# and Java) which has similar logic while parsing ResourceID generated by backend
    return base64.standard_b64decode(string_to_decode.replace('-', '/') + '==')
```
OS: Fedora Linux 22
Azure CLI Version: 2.0.46

**Comment by srinathnarayanan** [comment]
Hi @adisaraf , can we get a repro for this issue? You can ping us at askcosmosdb@microsoft.com with the repro. Is this consistently reproing? Thanks

**Comment by adisaraf** [comment]
Hi @srinathnarayanan Yes it is reproducible consistently. The reproduction steps are outlined above. 
```
az cosmosdb database show --db-name <db-name> --key <secondary key you provided> --url-connection <url>
```
Please let me know if you need anymore info.

**Comment by srinathnarayanan** [comment]
I was not able to repro it on a prod account. Can you please ping us on askcosmosdb@microsoft.com with the dbname, read key and url that you have used? Thanks.

**Comment by adisaraf** [comment]
Thanks @srinathnarayanan I have sent the details to that e-mail address.

**Comment by southpolesteve** [comment]
Closing this issue due to age. Active development of our Python SDK has moved to https://github.com/Azure/azure-sdk-for-python. We have also released 4.0 which includes many updates and new features. If this issue still needs our attention, please reopen it in the new repository.
